modules/Microsoft.HealthcareApis/workspaces/iotconnectors/deploy.bicep

```diff
@@ -0,0 +1,215 @@
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+var diagnosticsMetrics = [for metric in diagnosticMetricsToEnable: {
+  category: metric
+  timeGrain: null
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+var identityType = systemAssignedIdentity ? (!empty(userAssignedIdentities) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(userAssignedIdentities) ? 'UserAssigned' : 'None')
+
+var identity = identityType != 'None' ? {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
+} : null
+
+var enableReferencedModulesTelemetry = false
+
+// =========== //
+// Deployments //
+// =========== //
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource workspace 'Microsoft.HealthcareApis/workspaces@2022-06-01' existing = {
+  name: workspaceName
+}
+
+resource iotConnector 'Microsoft.HealthcareApis/workspaces/iotconnectors@2022-06-01' = {
+  name: name
+  parent: workspace
+  location: location
+  tags: tags
+  identity: identity
+  properties: {
+    ingestionEndpointConfiguration: {
+      eventHubName: eventHubName
+      consumerGroup: consumerGroup
+      fullyQualifiedEventHubNamespace: '${eventHubNamespaceName}.servicebus.windows.net'
+    }
+    deviceMapping: {
+      content: deviceMapping
+    }
+  }
+}
+
+resource iotConnector_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock)) {
+  name: '${iotConnector.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: iotConnector
+}
+
+resource iotConnector_diagnosticSettings 'Microsoft.Insights/diagnosticsettings@2021-05-01-preview' = if ((!empty(diagnosticStorageAccountId)) || (!empty(diagnosticWorkspaceId)) || (!empty(diagnosticEventHubAuthorizationRuleId)) || (!empty(diagnosticEventHubName))) {
+  name: diagnosticSettingsName
+  properties: {
+    storageAccountId: !empty(diagnosticStorageAccountId) ? diagnosticStorageAccountId : null
+    workspaceId: !empty(diagnosticWorkspaceId) ? diagnosticWorkspaceId : null
+    eventHubAuthorizationRuleId: !empty(diagnosticEventHubAuthorizationRuleId) ? diagnosticEventHubAuthorizationRuleId : null
+    eventHubName: !empty(diagnosticEventHubName) ? diagnosticEventHubName : null
+    metrics: diagnosticsMetrics
+    logs: diagnosticsLogs
+  }
+  scope: iotConnector
+}
+
+module fhir_destination 'fhirdestinations/deploy.bicep' = {
```

Comment:
Is this module not optional? If so, would it not be esier to have an input parameter `param fhirdestination object = {}` as it should move some of the input parameter complexity into the object?

And here use the header `module fhir_destination 'fhirdestinations/deploy.bicep' = if (!empty(fhirdestination)) {`

Also, can there only be one location?

Reply:
The destination is a child resource but there can only be one. YOu will get an error if you try to deploy more than 1.

Reply:
FHIR destination is optional per say. The IOT connector can be deployed without the destination, but the service won't do anything as the destination has not been set.

Reply:
Currently the only property a fhirDestination object would wrap is the destinationMapping object. I'm not sure I see a value in wrapping that since the IOT connector depends on that.